sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs

```diff
@@ -69,6 +89,12 @@ public class EventProcessor
         ///
         private ConcurrentDictionary<string, PartitionPump> PartitionPumps { get; }
 
+        /// <summary>
+        ///   The set of partition ownership this event processor owns.  Partition ids are used as keys.
+        /// </summary>
+        ///
+        private Dictionary<string, PartitionOwnership> InstanceOwnership { get; set; }
```

Comment:
Because we are always overwriting it in `RenewOwnership` method. If we instantiated it only in the constructor, we would need to keep clearing it and setting the keys one by one on every renew operation.